services/notifications/pkg/service/service.go

```diff
@@ -325,6 +310,37 @@ func (s eventsNotifier) handleShareCreated(e events.ShareCreated) {
 	}
 }
 
+func (s eventsNotifier) handleVirusscanFinished(e events.VirusscanFinished) {
+	if !e.Infected {
+		// no need to annoy the user
+		return
+	}
+
+	// should we send the email to all user who have access to the file? If so - how to find them?
+	if e.ExecutingUser == nil {
+		s.logger.Error().Str("events", "VirusscanFinished").Interface("resourceid", e.ResourceID).Str("uploadid", e.UploadID).Msg("no executing user")
+		return
+	}
+
+	msg, err := email.RenderEmailTemplate("virusdetected.email.tmpl", map[string]string{
+		"ExecutingUser": e.ExecutingUser.Username,
+		"Virus":         e.Description,
+		"Outcome":       string(e.Outcome),
+		"Filename":      e.Filename,
+	}, s.emailTemplatePath)
+	if err != nil {
+		s.logger.Error().Str("events", "VirusscanFinished").Interface("resourceid", e.ResourceID).Str("uploadid", e.UploadID).Msg("can't render email template")
+		return
+	}
+
+	senderDisplayName := "Owncloud"
+	emailSubject := "Virus detected"
```

Comment:
would it make sense to have it configurable?

Reply:
Templating sender/subject is another ticket. We should wait for master to have that implemented, then we can adjust templates for the Virusemail.